crates/core/src/doc/lives.rs

```diff
@@ -69,8 +70,10 @@ impl Document {
 			let initial = self.initial.doc.as_arc();
 			// Check if this is a delete statement
 			let doc = if stm.is_delete() {
+				self.computed_fields(stk, ctx, opt, DocKind::Initial).await?;
 				&self.initial
 			} else {
+				self.computed_fields(stk, ctx, opt, DocKind::Current).await?;
 				&self.current
 			};
```

Comment:
self.computed_fields is invoked inside the loop, recomputing the same document fields for every live query. Can we do that before the loop?